lms/lmsdb/models.py

```diff
@@ -365,6 +366,34 @@ def to_choices(cls: enum.EnumMeta) -> Tuple[Tuple[str, str], ...]:
         return tuple((choice.name, choice.value) for choice in choices)
 
 
+class SolutionEvaluation(BaseModel):
+    name = CharField()
+    icon = CharField(null=True)
+    color = CharField()
+    active_color = CharField()
+    order = IntegerField(default=0, index=True, unique=True)
+
+    @classmethod
+    def get_evaluations(cls):
+        return cls.select().order_by(cls.order)
+
+    def __str__(self):
+        return self.name
+
+
+@pre_save(sender=SolutionEvaluation)
+def evaluation_on_save_handler(_model_class, instance, created):
+    """Change colors to hex."""
+
+    is_color_changed = not instance.color.startswith('#')
+    if created or is_color_changed:
+        instance.color = COLORS.get(instance.color, '#0d6efd')
```

Comment:
Nice!

Another way that might prove useful:

```python
hex_color = re.compile(r'#?(?P<hex>[a-f0-9]{3}|[a-f0-9{6}])')

def get_hex(number: str) -> Optional[str]:
    if color := hex_color.match(number):
        return color.groupdict()['hex']
    raise ValueError("This is not a valid hex color.")


...
def evalutaion_on_save_handler etc
    try:
        instance.color = get_hex_color(...)
    except ValueError:
        instance.color = DEFAULT_PRIMARY_COLOR
```

Reply:
Also, to be really awesome, you might want to support input like "red", "blue" etc.
There's probably a really easy way to convert these words into hex RGB representation, and this can be awesome.

Reply:
I changed it like the way you did and added the `COLORS` dict in case the user entered a word like "red", "blue" and etc.